mosaic-runtime/src/commonMain/kotlin/com/jakewharton/mosaic/layout/KeyModifier.kt

```diff
@@ -0,0 +1,65 @@
+package com.jakewharton.mosaic.layout
+
+import androidx.compose.runtime.Immutable
+import com.jakewharton.mosaic.modifier.Modifier
+import dev.drewhamilton.poko.Poko
+
+public interface KeyModifier : Modifier.Element {
+	/**
+	 * This function is called when a [KeyEvent] is received by this node during the downward pass.
+	 * It gives ancestors of a focused component the chance to intercept an event. Return true to
+	 * stop propagation of this event. If you return false, the event will be sent to this
+	 * [KeyModifier]'s child. If none of the children consume the event, it will be sent
+	 * back up to the root using the [onKeyEvent] function.
+	 */
+	public fun onPreKeyEvent(event: KeyEvent): Boolean
+
+	/**
+	 * This function is called when a [KeyEvent] is received by this node during the upward pass.
+	 * While implementing this callback, return true to stop propagation of this event. If you
+	 * return false, the key event will be sent to this [KeyModifier]'s parent.
+	 */
+	public fun onKeyEvent(event: KeyEvent): Boolean
+}
+
+@[Immutable Poko]
+public class KeyEvent(
+	public val key: String,
```

Comment:
Is there a list of the names of all the keys somewhere?
I think it would be convenient to have a set of frequently used keys inside the library for quick and convenient use by client applications.

Reply:
This is not the final API. I don't like this API at all, actually. It's a direct mapping from Mordant, and that's enough to unblock users who need key input (there's at least one).

Long term a better API that has actual constants for keys and supports the full range of flags should be designed.

Reply:
Yeah I don't mean this necessarily as a criticism of your design choices, more that I'm not sure it makes sense for a relatively high-level UI abstraction like our Compose UI-based composables. We'll probably model ourselves _very_ close to what Compose UI is already doing just for the sake of familiarity.

I'm also just a little wary using strings as the domain of possible values as opposed to some closed set of constants. I know it's a bit tricky at your layer of abstraction because you receive readable text input in addition to key presses which can include multi-codepoint grapheme clusters like 🧑‍🌾.